generators/app/index.js

```diff
@@ -53,49 +111,80 @@ module.exports = generator.extend({
             poly.buildTool = conf.buildTool;
 
             // jhipsterFunc polyfill :
-            poly.replaceContent = () => jhipsterModuleSubgenerator.prototype.replaceContent;
             poly.registerModule = jhipsterModuleSubgenerator.prototype.registerModule;
             poly.updateEntityConfig = jhipsterModuleSubgenerator.prototype.updateEntityConfig;
 
             // @todo : handle this.options.testMode ?
 
             return poly;
         }, onError => console.error(onError));
-    },
+    }
 
     /**
-     * replace Spring naming strategies with more neutral ones
-     * return true if all occurrences are replaced
+     * Replace Spring naming strategies with more neutral ones.
      *
-     * note : after running this function, reference to the ancient naming strategies will still be found in :
-     * ./node_modules/generator-jhipster/generators/server/templates/_pom.xml:
+     * Note : after running this function, a reference to the ancient naming strategies will still be found in :
+     * ./node_modules/generator-jhipster/generators/server/templates/_pom.xml
      * however this doesn't concern us
-     *
-     * @todo : write local test for the return value
-     * @todo : write unit test
-     * @todo : no hardcoded values for removeGradleFiles and removeMavenFiles
      */
-    _replaceNamingStrategies: (appBuildTool) => {
+    _replaceNamingStrategies (appBuildTool) {
         const physicalOld = DBH_CONSTANTS.physicalNamingStrategyOld;
         const physicalNew = DBH_CONSTANTS.physicalNamingStrategyNew;
 
         const implicitOld = DBH_CONSTANTS.implicitNamingStrategyOld;
         const implicitNew = DBH_CONSTANTS.implicitNamingStrategyNew;
 
+        // depending on the application's build tool, get all files where the old naming strategies must be replaced
         const files = dbh.getFilesWithNamingStrategy(appBuildTool);
 
+        // return an absolute path in order to make debugging easier
+        const getFilesAbsolutePath = (filesArray) => {
+            let arr = [];
+
+            // will be true if a testCase has been passed
+            if (this.dbhTestCase) {
+                filesArray.forEach((value) => {
+                    arr.push(path.join(DBH_TEST_CONSTANTS.testConfigDir[this.dbhTestCase], value));
+                });
+            }
+
+            else {
+                filesArray.forEach((value) => {
+                    arr.push(path.join(__dirname, value));
+                });
+            }
+
+            return arr;
+        };
+
+        const filesWithAbsolutePath = getFilesAbsolutePath(files);
+
         // check that each file exists, then replace the naming strategies
         files.forEach((path) => {
             if (fs.existsSync(path)) {
                 // 1) replace Spring physical naming strategy
-                jhipsterFunc.replaceContent(path, physicalOld, physicalNew);
+                this._replaceContent(path, physicalOld, physicalNew);
```

Comment:
We currently don't have `fix-entity/index.js` as parameter for a yeoman-test RunContext object. That's why we don't have a polyfill at the moment. But now that I have an idea of how things work, the same logic can be implemented again. Hopefully with much less suffering.